Server/src/controllers/DataSetController.ts

```diff
@@ -0,0 +1,245 @@
+            } catch (err) {
+                response.status(500).send(err);
+            }
+        }
+    }
+
+    /**
+     * This controller will take a request, grab the user ID, and if it a real number will then process the request,
+     * send it to the getDataService to acquire an array of data sets that were saved by this user ID, and
+     * then return a response with the array of data sets. If the request is invalid, or some other problem arrises, 
+     * it will throw an error.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForUserSavedDatsets(request: Request, response: Response) {
+        let userId: any = request.body.user.account_id
+        if (isNaN(userId)) {
+            response.status(500).json("Invalid search params entered");
+        }
+        else {
+            try {
+                let arrayOfData = await this.dataSetService.getUserSavedDatasets(userId)
+                return response.status(200).send(arrayOfData);
+            } catch (err) {
+                response.status(500).send(err);
+            }
+        }
+    }
+
+    //TODO: Readd later
+    /**
+     * This controller will take a request, grab the user email and data set ID, verify said ID is a number,
+     * and then send the service a request to add the user's saved data set preference to the database.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+    async createRequestForAddingSavedDataset(request: Request, response: Response) {
+        let userEmail = request.params.userEmail;
+        let requestParam = request.params.datasetId;
+        let datasetId: number = +requestParam;
+        if (isNaN(datasetId)) {
+            response.status(400).json("Invalid data set ID entered");
+        }
+        else {
+            try {
+                let executionStatus = await this.dataSetService.addSavedDatasetService(userEmail, datasetId)
+                if (executionStatus[0]) { return response.status(200).json(executionStatus[1]); }
+                else { return response.status(400).json(executionStatus[1]); }
+            } catch (err) {
+                response.status(500).json(err);
+            }
+        }
+    }
+     */
+
+    /**
+     * This controller will take a request, send it to the getDataService to acquire an array 
+     * containing the unapproved data sets in the database
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForUnapprovedDatsets(request: Request, response: Response) {
+        try {
+            this.dataSetService = new DataSetService();

```

Comment:
this is what i mean, here a new instance is created, but in other cr3eateRequests method its not, are they guaranteed to have datsetService instantiated?

Reply:
It is missing from the other methods.... Was forgotten to add. Good catch

Reply:
Added to a number of methods